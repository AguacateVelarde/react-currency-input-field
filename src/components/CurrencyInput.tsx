import React, { PureComponent } from 'react';

import {
  addCommas,
  removeCommas,
  checkIsValidNumber,
} from './utilities';

interface IProps {
  id: string;
  className?: string;
  limit?: number;
  prefix?: string;
  handleError?: () => void,
  onChange: (value: number) => void;
  placeholder?: string;
}

interface IState {
  value: string;
}

export class CurrenyInput extends PureComponent<IProps, IState> {
  private constructor(props: IProps) {
    super(props);

    this.state = {
      value: '',
    };

    this.processChange = this.processChange.bind(this);
  }

  processChange(event: React.ChangeEvent<HTMLInputElement>) {
    const {
      target: {
        value,
      }
    } = event;

    const {
      onChange,
      limit,
      prefix,
    } = this.props;

    let stringValue = value;

    if (prefix) {
      stringValue = value.replace(prefix, '');
    }

    if (stringValue === '' || stringValue === null) {
      this.setState({
        value: '',
      });
      onChange(null);
      return false;
    }

    let intValue = parseInt(removeCommas(stringValue), 10);

    const max = limit || 9999999999999;

    if (checkIsValidNumber(intValue, max)) {
      let setValue = addCommas(intValue);

      if (prefix) {
        setValue = `${prefix}${setValue}`;
      }

      this.setState({ value: setValue });
    }
    onChange(intValue);
  }

  render() {
    const {
      className,
      id,
      handleError,
      placeholder,
    } = this.props;

    const { value } = this.state;

    return (
      <input
        type="string"
        id={id}
        className={className}
        onChange={this.processChange}
        placeholder={placeholder}
        value={value}
      />
    );
  }
}

export default CurrenyInput;
