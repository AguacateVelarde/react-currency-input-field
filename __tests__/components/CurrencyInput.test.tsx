import { shallow } from 'enzyme';
import React from 'react';
import CurrencyInput from '../../src/components/CurrencyInput';

describe('<CurrencyInput /> component', () => {
  test('Renders without error', () => {
    const view = shallow(
      <CurrencyInput
        id="validationCustom01"
        placeholder="Â£1,000"
        className="form-control"
        onChange={() => {}}
      />
    );
    const component = view.find('#settings__checkbox');
    expect(component).toMatchSnapshot();
  });
});
