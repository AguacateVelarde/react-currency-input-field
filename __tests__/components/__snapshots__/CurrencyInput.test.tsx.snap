// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyInput /> component Renders without error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <CurrenyInput
      className="form-control"
      id="validationCustom01"
      onChange={[Function]}
      placeholder="£1,000"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "form-control",
        "id": "validationCustom01",
        "onChange": [Function],
        "placeholder": "£1,000",
        "type": "string",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "id": "validationCustom01",
          "onChange": [Function],
          "placeholder": "£1,000",
          "type": "string",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
